client/asset/btc/btc.go

```diff
@@ -2270,6 +2291,37 @@ func (btc *baseWallet) AuditContract(coinID, contract, txData dex.Bytes, rebroad
 	}, nil
 }
 
+// AuditContract retrieves information about a swap contract from the provided
+// txData. The extracted information would be used to audit the counter-party's
+// contract during a swap. Since this wallet is backed by a full node, txData
+// may be empty to attempt retrieval of the transaction output from the network.
+func (btc *ExchangeWalletFullNode) AuditContract(coinID, contract, txData dex.Bytes, rebroadcast bool) (*asset.AuditInfo, error) {
+	if len(txData) != 0 {
+		return btc.baseWallet.AuditContract(coinID, contract, txData, rebroadcast)
+	}
+
+	full, ok := btc.node.(*rpcClient)
+	if !ok {
+		return nil, fmt.Errorf("wallet backend not a *rpcClient: %T", btc.node)
+	}
+
+	// In the off chance that the transaction is in mempool or txindex is
+	// enabled, try getrawtransaction.
+	txHash, _, err := decodeCoinID(coinID)
+	if err != nil {
+		return nil, err
+	}
+	tx, err := full.GetRawTransaction(txHash)
+	if err != nil {
```

Comment:
This call is a little funny, because, as you mentioned, it will only work with mempool or txindex, but what we're really after is just the `*wire.TxOut`, and `gettxout` works for mempool and doesn't require txindex. The only difference is that now the tx will be rebroadcast, but that's primarily a concern for SPV clients who can't see mempool transactions, and at this point, we already know we're not SPV. You've added code to `(*baseWallet).AuditContract` to do the `gettxout` call already, and `rpcClient` already ignores the 48-hour filter and passes `true` for `includeMempool`. From my read, it looks like the entire `(*ExchangeWalletFullNode).AuditContract` method could be foregone without consequence.

Reply:
Huh, quite right.  If the `getrawtransaction` succeeds, there's no point in rebroadcasting it because the node would reject it as already existing.